### PR TITLE
Loadout Systems for players

### PR DESCRIPTION
**Pull request by MidnightGH**

### CODE REVIEW

File: [MAAS].Tanoa/functions/player/loadouts/fn_applyLoadout.sqf [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  Author: Midnight
+
+  Description: Applies loadout from config to given unit
+
+  Paramter(s):
+  0 - OBJECT: Unit to apply loadout config to
+  1 - STRING: classname of loadout to be applied (mission config)
+*/
+params[["_unit",objNull,[objNull],1],["_config","",[""],1]];
+private _load = (missionConfigFile >> "CfgLoadouts" >> "maas");
```
Comment (dedmen) [inline [MAAS].Tanoa/functions/player/loadouts/fn_applyLoadout.sqf]: Why do you private this variable but not the rest? You should private all of them.

File: [MAAS].Tanoa/functions/player/loadouts/fn_applyLoadout.sqf [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+_name = getText(_load >> _config >> "name");
```
Comment (dedmen) [inline [MAAS].Tanoa/functions/player/loadouts/fn_applyLoadout.sqf]: Make a macro.
```
#define GET_CONFIG_PARAM_TEXT(target, name) private target = getText (_load >> _config >> #name)
#define GET_CONFIG_PARAM_NUM(target, name) private target = getNumber (_load >> _config >> #name)
#define GET_CONFIG_PARAM_ARR(target, name) private target = getArray (_load >> _config >> #name)

GET_CONFIG_PARAM_TEXT(_name, name);
GET_CONFIG_PARAM_TEXT(_uniform, uniform);
...
GET_CONFIG_PARAM_NUM(_secondaryMC, secondaryMagCount);
GET_CONFIG_PARAM_ARR(_secondaryItems, secondaryWeaponItems);
```

File: [MAAS].Tanoa/functions/player/loadouts/fn_applyLoadout.sqf [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+_primaryItems = getArray(_load >> _config >> "primaryWeaponItems");
+_secondaryItems = getArray(_load >> _config >> "secondaryWeaponItems");
+_fakCount = getNumber(_load >> _config >> "FAKCount");
+_map = getNumber(_load >> _config >> "map");
+_compass = getNumber(_load >> _config >> "compass");
+_gps = getNumber(_load >> _config >> "gps");
+_watch = getNumber(_load >> _config >> "watch");
+_pMag = getArray(configFile >> "CfgWeapons" >> _primary >> "magazines") select 0;
+_sMag = getArray(configFile >> "CfgWeapons" >> _secondary >> "magazines") select 0;
+{_unit unlinkItem _x;} forEach assignedItems _unit;
```
Comment (dedmen) [inline [MAAS].Tanoa/functions/player/loadouts/fn_applyLoadout.sqf]: clear entire loadout?
`_unit setUnitLoadout (getUnitLoadout objNull)`
AFAIK `(getUnitLoadout objNull)` will return a empty loadout. If it doesn't just use `[]` that should also work.

File: [MAAS].Tanoa/functions/player/loadouts/fn_applyLoadout.sqf [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+_unit addBackpack _backpack;
+_unit addHeadgear _headgear;
+_unit addMagazines[_pMag,_primaryMC];
+_unit addMagazines[_sMag,_secondaryMC];
+_unit linkItem _goggles;
+_unit linkItem _hmd;
+{_unit addPrimaryWeaponItem _x} forEach _primaryItems;
+{_unit addSecondaryWeaponItem _x} forEach _secondaryItems;
+for "_i" from 0 to _fakCount do {_unit addItem "FirstAidKit";};
+if(_map == 1) then {_unit linkItem "ItemMap";};
```
Comment (dedmen) [inline [MAAS].Tanoa/functions/player/loadouts/fn_applyLoadout.sqf]: the `;` inside the {} isn't needed.
you could make another 
`#define GET_CONFIG_PARAM_BOOL(target, name) GET_CONFIG_PARAM_NUM(target, name) != 0`
to get a boolean and not have the `== 1` everywhere here.

File: [MAAS].Tanoa/functions/player/loadouts/fn_applyLoadout.sqf [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+if(_compass == 1) then {_unit linkItem "ItemCompass";};
+if(_gps == 1) then {_unit linkItem "ItemGPS";};
+if(_watch == 1) then {_unit linkItem "ItemWatch";};
+reload _unit;
```
Comment (dedmen) [inline [MAAS].Tanoa/functions/player/loadouts/fn_applyLoadout.sqf]: if you add the magazines before you add the weapons they will automatically take a magazine and reload themselves.

File: [MAAS].Tanoa/config/CfgLoadouts.hpp [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+      goggles="";
+      hmd="O_NVGoggles_ghex_F";
+      uniform="U_O_T_Soldier_F";
+      vest="V_HarnessO_ghex_F";
+      backpack="";
+      primary="arifle_CTAR_blk_F";
+      secondary="hgun_Rook40_F";
+      launcher="";
+      primaryMagCount=2;
+      primaryWeaponItems[]={};
```
Comment (dedmen) [inline [MAAS].Tanoa/config/CfgLoadouts.hpp]: You have 2 magazines. But 0 magazine classnames?

Comment (MidnightGH) [reply [MAAS].Tanoa/config/CfgLoadouts.hpp]: applyLoadout fnc detects given magazine for primary and secondary. However, I'm going to add a entry for extra magazines & if the given weapon class should receive it's respective magazine set.

**Pull request merged**